extensions/spatial.sigmf-ext.md

```diff
@@ -0,0 +1,362 @@
+is illustrated in Figure 1 below. The ISO 80000-2:2019 compliant cartesian and
+spherical systems have a specific relationship with the reported azimuth and
+elevation, which use the conventional geospatial definitions (degrees east of
+north, and degrees above horizon respectively).
+
+![SigMF Spatial Coordinate Reference System](spatial_crs.png)
+
+**Figure 1 - SigMF Spatial Extension Coordinate Reference Systems**
+
+As shown in Figure 1, the boresight of the aperture is defined as being in the
+direction of the positive X-axis. Based on this, as an example, a horizontal
+uniform linear array aperture would be defined along the Y-axis (see examples
+section below).
+
+## 0 Datatypes
+
+This extension defines the following datatypes:
+
+|type|long-form name|description|
+|----|--------------|-----------|
+|bearing|signal direction bearing|Quantitative representation of a direction with optional error.|
+|cartesian_point|cartesian position|Cartesian coordinate triplet [x, y, z]|
+
+### 0.1 The `bearing` Object
+
+The `bearing` field is used to describe relative one or two dimensional signal
+directions. The units are always in degrees; while it is legal for this field to
+have any value it is RECOMMENDED that it be between 0 and 360 or +/- 180.
+
+|name|required|type|units|description|
+|----|--------|----|-----|-----------|
+|`azimuth`|false|float|degrees|Azimuth component of the direction in degrees increasing clockwise.|
+|`elevation`|false|float|degrees|Elevation component of the direction in degrees above horizon.|
+|`range`|false|float|meters|Line-of-sight slant range to emitter, if known, in meters.|
+|`az_error`|false|float|degrees|Error or uncertainty in the azimuth component.|
+|`el_error`|false|float|degrees|Error or uncertainty in the elevation component.|
+|`range_error`|false|float|meters|Error or uncertainty in the range component.|
+
+The `az_error`, `el_error`, and `range_error` fields SHOULD NOT be used if the
+corresponding field estimates are not included.
+
+```json
+  "bearing": {
+    "azimuth": 211.2,
+    "elevation": 15.0,
+    "range": 30,
+    "az_error": 0.1,
+    "el_error": 0.1,
+    "range_error": 0.1
+  }
+```
+
+### 0.2 The `cartesian_point` Object
+
+The `cartesian_point` field is used to describe a single point in the Cartesian
+coordinate reference system. This object is necessary to define the phase center
+geometries for multidimensional arrays, and other cartesian locations.
+
+|name|required|type|units|description|
+|----|--------|----|-----|-----------|
+|`point`|true|float array|meters|Cartesian Point defined by a element array of [x,y,z] referenced to the `spatial` CRS.|
+
+
+## 1 Global
+
+The `spatial` extension adds the following fields to the `global` SigMF object:
+
+|name|required|type|units|description|
+|----|--------|----|-----|-----------|
+|`num_elements`|true|int|N/A|Defines the number of phase centers / channels collected in the Collection or multichannel Dataset.|
```

Comment:
Would this be redundant to some description in the `.sigmf-collection`? Or, among the various `.sigmf-meta` files in a collection? (since it would appear in each).
In the case of a multi-channel `.sigmf-data`, it's redundant to `num_channels`, isn't it?
If you're trying to bring multi-channel collections and multi-channel `.sigmf-data` under a single umbrella, I can see some benefit in that, but there's a price for the redundancy. The data *processing* (or, handling) of those two different scenarios are *very* different (especially if you don't have consistent time stamps). That makes me wonder how valuable this common field really is.

Reply:
This extension predated formal multichannel and collection support, so some of this is a bit convoluted now. It is redundant in some situations but i'm not sure how to make this better. Collections dont specify the number of channels; the length of the `core:streams` array can be a proxy but its not guaranteed that all recordings in a collection are coherent channel data streams, nor is it guaranteed they are all used.

Conditional requirement is not something SigMF supports, and Collections are the primary way this extension is used so "required" was opted for. It is redundant for multichannel, single file datasets.

Datasets with differing timestamps or offsets should not be an issue for properly functioning processing software.

Reply:
In addition to the idea that not all streams in a collection represent spatially associated (coherent) data, a collection should allow for multiple separately sets of coherent data. For example a collection could contain (real example BTW):
- `geolocation.sigmf-meta` (time-changing position of collector)
- `gimbal_position.sigmf-meta` (time-changing bearing of array)
- `observation_stream_ch0.sigmf-meta` (500MS/s observation data RX0)
- `observation_stream_ch1.sigmf-meta` (500MS/s observation data RX1)
- `main_beam_ch0.sigmf-meta` (30.72MS/s primary aperture data RX0.0)
- `main_beam_ch1.sigmf-meta` (30.72MS/s primary aperture data RX0.1)
...
- `main_beam_ch7.sigmf-meta` (30.72MS/s primary aperture data RX3.1)

Specifying this explicitly in the collection _might_ help in some simpler cases, but this is only redundant in simple cases, and ultimately I think that the cost of a single redundant field is negligible especially considering it contains contextually useful information even in a standalone sense.